kyma-project/kyma
# Add update helm chart functionality for charts already deployed

Related to #1294 

**Proposed Steps to test Pull Request**

1) Install heptio ark from https://github.com/heptio/ark/releases
2) Add ark backup resources (download file and change to yaml [ark_backup.txt](https://github.com/kyma-project/kyma/files/2722878/ark_backup.txt))
   `kubectl create -f ark_backup.yaml`
3) If you work on minikube find minikube IP and minio service in `heptio-ark` namespace
   `kubectl get service -n heptio-ark`
4) Add resource from content:
    ```
    apiVersion: ark.heptio.com/v1
    kind: BackupStorageLocation
    metadata:
      name: default
      namespace: heptio-ark
    spec:
      provider: aws
      objectStorage:
        bucket: ark
      config:
        region: minio
        s3ForcePathStyle: "true"
        s3Url: http://<minikube_ip>:<minio_service_port>
    ```
    where `minikube_ip` and `minio_service_port` you got in point 3)
5) Create ServiceInstance according to https://github.com/kyma-project/examples/blob/master/service-binding/lambda/README.md
   If you create ServiceBinding you need to be aware of issue connected with secrets occurs after restore backup install ark plugin.
   The issue is described here: https://github.com/heptio/ark/issues/816 and to resolve it you need to install ark-plugin from
   https://github.com/kyma-project/kyma/tree/master/tools/ark-plugins by adding plugin to you ark
   `ark plugin add eu.gcr.io/kyma-project/develop/ark-plugins`
6) Create backup
   `ark backup create cluster-backup --kubeconfig=/path/to/kubeconfig`
   check if backup is already done `ark backup get`
   in browser go to <minikube_ip>:<minio_service_port> and login to minio with `minio` and `minio123` credentials
   download created backup files
7) Remove cluster and install Kyma again
8) Repeat points from 2 to 4
9) Restore created backup `ark restore create --from-backup cluster-backup --kubeconfig=/path/to/kubeconfig`
   and check if restore is in completed state `ark restore get`

   Quick way to restore is e.g. create backup for only one namespace
   `ark backup create cluster-backup --include-namespaces=production --kubeconfig=/path/to/kubeconfig`, remember backup name need to be the same as before
   then go to <minikube_ip>:<minio_service_port>, then go to created backup, remove files and replace them with files with you created before.
